### ISSUE TITLE
we should not suggest code completions for dartdoc comments

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/sdk#34747]
Currently, we suggest completions in dartdoc comments. We should change this to not suggest for general dartdoc (but should still offer suggestions for symbol references in dartdoc, like `[FooClass]`).

<img width="251" alt="screen shot 2018-10-10 at 12 56 20 pm" src="https://user-images.githubusercontent.com/1269969/46762449-ecca4700-cc8b-11e8-895a-6cc366bb42cf.png">

**Comment by scheglov** [comment]
Please provide a code sample where it happens.
I cannot reproduce it locally.

```
1539203266397 => {"id":"13995","method":"completion.getSuggestions","params":{"file":"/Users/scheglov/Source/Dart/sdk.git/sdk/pkg/analyzer/lib/src/dart/element/inheritance_manager2.dart","offset":525},"clientRequestTime":1539203266397}
1539203266397 <= {"id":"13995","result":{"id":"672"}}
1539203266397 <= {"event":"completion.results","params":{"id":"672","replacementOffset":525,"replacementLength":0,"results":[],"isLast":true}}
```

But IntelliJ shows when I press `Ctrl+Space`.

**Comment by DanTup** [comment]
VS Code has a setting `editor.wordBasedSuggestions` which causes this. It's on by default, but it really sucks. I have an incomplete "recommended settings" page for the Dart Code website I keep meaning to finish (basically a list of what I would consider bad defaults!).

If it was VS Code where you saw this, it'd be good to know if disabling this setting fixes it for you.

**Comment by devoncarew** [comment]
Yup, that fixed it for VS Code. We should recommend that setting :)

**Comment by DanTup** [comment]
Cool; I've fleshed out the Recommended Settings page a little and pushed it to a branch for input before I merge -> https://github.com/Dart-Code/Website/pull/2/files
